pkg/processor/trigger/kafka/event.go

```diff
@@ -41,3 +43,31 @@ func (e *Event) GetShardID() int {
 func (e *Event) GetPath() string {
 	return e.kafkaMessage.Topic
 }
+
+func (e *Event) GetTimestamp() time.Time {
+	return e.kafkaMessage.Timestamp
+}
+
+func (e *Event) GetHeaders() map[string]interface{} {
+	return e.getHeadersAsMap()
+}
+
+// GetHeader returns the header by name as an interface{}
+func (e *Event) GetHeader(key string) interface{} {
+	return e.getHeadersAsMap()[key]
+}
+
+func (e *Event) getHeadersAsMap() map[string]interface{} {
+	headersMap := map[string]interface{}{}
+
+	if e.kafkaMessage.Headers == nil {
+		return headersMap
+	}
+
+	// iterate over the header records and add each key-value to the map
+	for _, headerRecord := range e.kafkaMessage.Headers {
+		headersMap[string(headerRecord.Key)] = string(headerRecord.Value)
```

Comment:
keys to string - I get
Why not leave the native Value type though? why stringify?

Reply:
good point..no reason to stringify the value